Refactor to finder resource

ryanb-gds opened the pull request:
Refactoring to use a resource route collection for all endpoints relating to finders should make leveraging Rails conventions easier, hopefully increasing our future productivity when working on Specialist Publisher.

This work is being done as part of https://trello.com/c/lRu3iVj3

ryanb-gds committed:
Created finders resource route

This is the first step in refactoring the various routes that deal with finders into a cohesive finders resource

ryanb-gds committed:
Refactor request new finder code to finders resource

We are gradually refactoring finder related routes to live under a finder resource, in order to make easier use of Rails conventions.

Rather than duplicate the facet and metadata form partials, we will load them from the admin directory for now before moving them across in a later commit, to make the diff more reviewable.

ryanb-gds committed:
Moved finder summary endpoint into finders resource

As part of this refactor, I moved the authorisation test down the pyramid into the controller test, as checking authorisation using a feature test seems heavy-handed.

ryanb-gds committed:
Extend access to finders index to non GDS users

This page should be available to all users. Users will only see it as the root page of the app if they can preview the design system.

As part of this commit, we tidy up the finder schema policy and its usage, so that we're making better use of the conventions provided by the Pundit gem we use for authorisation

ryanb-gds committed:
Move the zendesk ticket creation route from admin controller to finder controller

We can now remove the admin controller, as this was the last route to move to the finder controller

ChrisBAshton commented on app/views/finders/_facet_fields.html.erb:
Is there any way of restructuring the commit so that we effectively have a 'rename' of the existing https://github.com/alphagov/specialist-publisher/blob/cb937509d94a4244ca63c0e18fe4f7dd5bf3f054/app/views/admin/_facet_fields.html.erb file, rather than duplicating the file contents? It would be nice to know for certain that the file contents are identical (and where the code came from) rather than making the assumption as a reader.
E.g. by specifying template override in the controller until the final commit where we delete the admin controller - can then add one new 'final' commit that removes the override and renames the templates.

If not, perhaps clarify in the commit message that you'd copied the code over from that file, with a link to the sha for posterity.

ryanb-gds replied:
Hadn't thought of that. I'll have a look

ryanb-gds replied:
I have done this, but slightly stuffed up and accidentally merged the facets and metadata commits accidentally at some point. Not the end of the world, but rebasing is a tricky business 😅

ChrisBAshton reviewed (approved):
Looks good 👍 one non-blocking suggestion below, which would make it easier to review commit-by-commit

ryanb-gds committed:
Move finder template partials into correct directory

For neater git commit diffing purposes, we have been referencing finder template partials within the original admin view directory whilst refactoring the admin routes to a finders resource. Now that we have completed the refactoring, the partials can be moved across too.

ryanb-gds committed:
Move finder metadata and facet editing routes on to finders resource

This commit also renames the actions from "confirm" to "update", in order to better adhere to Rails resource conventions. We may choose to move metadata and facets on to their own resources in the future.
